pkg/backend/display/progress.go

```diff
@@ -1236,63 +1263,79 @@ func (display *ProgressDisplay) getStepDoneDescription(step engine.StepEventMeta
 	}
 
 	getDescription := func() string {
+		opText := ""
 		if failed {
 			switch op {
 			case deploy.OpSame:
-				return "failed"
+				opText = "failed"
 			case deploy.OpCreate, deploy.OpCreateReplacement:
-				return "creating failed"
+				opText = "creating failed"
 			case deploy.OpUpdate:
-				return "updating failed"
+				opText = "updating failed"
 			case deploy.OpDelete, deploy.OpDeleteReplaced:
-				return "deleting failed"
+				opText = "deleting failed"
 			case deploy.OpReplace:
-				return "replacing failed"
+				opText = "replacing failed"
 			case deploy.OpRead, deploy.OpReadReplacement:
-				return "reading failed"
+				opText = "reading failed"
 			case deploy.OpRefresh:
-				return "refreshing failed"
+				opText = "refreshing failed"
 			case deploy.OpReadDiscard, deploy.OpDiscardReplaced:
-				return "discarding failed"
+				opText = "discarding failed"
 			case deploy.OpImport, deploy.OpImportReplacement:
-				return "importing failed"
+				opText = "importing failed"
+			default:
+				contract.Failf("Unrecognized resource step op: %v", op)
+				return ""
 			}
 		} else {
 			switch op {
 			case deploy.OpSame:
-				return ""
+				opText = ""
 			case deploy.OpCreate:
-				return "created"
+				opText = "created"
 			case deploy.OpUpdate:
-				return "updated"
+				opText = "updated"
 			case deploy.OpDelete:
-				return "deleted"
+				opText = "deleted"
 			case deploy.OpReplace:
-				return "replaced"
+				opText = "replaced"
 			case deploy.OpCreateReplacement:
-				return "created replacement"
+				opText = "created replacement"
 			case deploy.OpDeleteReplaced:
-				return "deleted original"
+				opText = "deleted original"
 			case deploy.OpRead:
 				// nolint: goconst
-				return "read"
+				opText = "read"
 			case deploy.OpReadReplacement:
-				return "read for replacement"
+				opText = "read for replacement"
 			case deploy.OpRefresh:
-				return "refresh"
+				opText = "refresh"
 			case deploy.OpReadDiscard:
-				return "discarded"
+				opText = "discarded"
 			case deploy.OpDiscardReplaced:
-				return "discarded original"
+				opText = "discarded original"
 			case deploy.OpImport:
-				return "imported"
+				opText = "imported"
 			case deploy.OpImportReplacement:
-				return "imported replacement"
+				opText = "imported replacement"
+			default:
+				contract.Failf("Unrecognized resource step op: %v", op)
+				return ""
 			}
 		}
+		start, ok := display.opStopwatch.start[step.URN]
+		if !ok {
+			return opText
+		}
 
-		contract.Failf("Unrecognized resource step op: %v", op)
-		return ""
+		end, ok := display.opStopwatch.end[step.URN]
+		if !ok {
+			return opText
```

Comment:
Wander if we should add a bit that the timing information was missing? It shouldn't ever be hit but if it does might prompt people to tell us about it.

Reply:
I think silently erroring is fine in this case. It could throw off the progress display to print lines in this codepath or scare the user on a warning too much.

Someone will probably report a bug that a resource's time elapsed isn't showing up when all of the others are showing up.

Reply:
Yeh thinking on this more I'd be happier with a rework to make the missing times unrepresentable, but that's a minor thing we can pick up when it's convenient.